### PR TITLE
Feature/categories

### LINKED ISSUE #12: Additional Navigation on main page
This drawer should make the desktop version look less empty. 
1. Here we should think how we want to sort environments such that a side bar makes sense.
2. We could also add a short "About SciGym" and the link to an Impressum here (see #10)

### PR DESCRIPTION
**Pull request by HendrikPN**
This PR should close #12 and #8 
However, the Drawer in Home is not yet mobile friendly. In the future, we want it to be swipable (see material-ui [Drawer](https://material-ui.com/demos/drawers/#swipeable-temporary-drawer) component).

### CODE REVIEW

File: src/components/home/TopicDrawer.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    this.state = {
+      open: Array.apply(null, Array(topTopics.length)).map(function() {
+        return false;
+      }),
+      error: '',
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.topics !== prevProps.topics) {
```
Comment (erik-sn) [inline src/components/home/TopicDrawer.js]: I'm not sure this is returning what you are expecting - this is comparing a list of objects with equality which is not the same as comparing if the two lists contain the same objects. See:

https://stackoverflow.com/questions/7837456/how-to-compare-arrays-in-javascript

File: src/components/home/TopicDrawer.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+      const topTopics = this.props.topics.filter(topic => !topic.parentTopic);
```
Comment (erik-sn) [inline src/components/home/TopicDrawer.js]: I might rename this to `parentTopics` to make it more apparent that this `top` is a hierarchy, not top as in most popular.

File: src/components/home/TopicDrawer.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      this.setState({
+        open: Array.apply(null, Array(topTopics.length)).map(function() {
```
Comment (erik-sn) [inline src/components/home/TopicDrawer.js]: You can change this to `Array.apply(null, Array(topTopics.length)).map(() => false)`.

File: src/components/home/TopicDrawer.js [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    //TODO: handle FAILURE
+    if (id === 'all') {
+      this.props.resetSearchedEnvironments();
+    } else this.props.searchEnvironmentsByTopic(id);
+  };
+
+  render() {
+    const { classes, topics } = this.props;
+
+    const topTopics = topics.filter(topic => !topic.parentTopic);
```
Comment (erik-sn) [inline src/components/home/TopicDrawer.js]: again might change this to `parentTopics` and `childTopics`

File: src/components/profile/EnvironmentForm.jsx [hunk 5]
```diff
@@ -47,6 +49,7 @@ class EnvironmentForm extends Component {
       tag: '',
       tags: props.envExists && Boolean(props.environment.tags) ? props.environment.tags : [],
       error: '',
+      topic: props.envExists && Boolean(props.environment.topic) ? props.environment.topic.id : '',
```
Comment (erik-sn) [inline src/components/profile/EnvironmentForm.jsx]: Just to clean this up a little I might deconstruct `props` so you don't have to repeatedly access it:
```
const { envExists, environment} = props;
this.state = {
  ...
}
```

File: src/components/home/TopicDrawer.js [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+                <ListItem
+                  button
+                  key={topTopic.id}
+                  onClick={() => this.handleTopClick(index, topTopic.id)}
+                >
+                  <ListItemText primary={topTopic.name} />
+                  {this.state.open[index] ? <ExpandLess /> : <ExpandMore />}
+                </ListItem>
+                <Collapse in={this.state.open[index]} timeout="auto" unmountOnExit>
+                  {botTopics.map(
```
Comment (erik-sn) [inline src/components/home/TopicDrawer.js]: When a `render` function gets this long it's usually a sign that it should be broken up into different components. For example the JSX in these nested map function could be moved to it's own component.

**Review by erik-sn — CHANGES_REQUESTED** [review]
Left some comments!

File: src/components/home/TopicDrawer.js (hunk 1 above)
Comment (HendrikPN) [reply src/components/home/TopicDrawer.js]: This should not matter. The topics are static, this is just to make sure that it recognizes that there are topics as opposed to no topics at all. Hence, this should suffice. Or do you disagree.

Comment (HendrikPN) [reply src/components/home/TopicDrawer.js]: Actually, you are right, I changed it to
```
if (this.props.topics.length > prevProps.topics.length) {
...
}
```

**Review by erik-sn — APPROVED** [review]

**Pull request merged**